### PR TITLE
Add models and migrations

### PR DESCRIPTION
**Pull request by Dezash**
https://kamandax.atlassian.net/browse/MOB-55

Added all models according to the database structure found here:
https://github.com/KamandaX/Wiki/blob/master/DatabaseStructures.md

This requires manually setting up database credentials with app-secrets. Here are the instructions on how to do that:
https://github.com/KamandaX/Wiki/pull/8

For testing purposes, I've added SeedData class, which will execute if the database is empty and generate some mockup data.

### CODE REVIEW

File: API/Models/ApiContext.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
```
Comment (wizacass) [inline API/Models/ApiContext.cs:5]: Unnecessary usings.

File: API/Models/Camera.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+using System.Threading.Tasks;
```
Comment (wizacass) [inline API/Models/Camera.cs:7]: Unnecessary usings.

File: API/Models/SeedData.cs [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                },
+                new QuestionOption
+                {
+                    ID = 5,
+                    QuestionID = 5,
+                    OptionContent = "stars",
+                    QuestionMultiplier = 0.5f
+                }
+            );
+            #endregion
```
Comment (wizacass) [inline API/Models/SeedData.cs:221]: Why this data here is hardcoded?

File: API/Startup.cs [hunk 4]
```diff
@@ -21,15 +24,27 @@ public void ConfigureServices(IServiceCollection services)
         {
             services.AddSingleton<Iserializer, JsonSerializer>();
             services.AddSingleton<IErrorFormatter, JsonErrorFormatter>();
+            //services.AddDbContext<ApiContext>(opt => opt.UseInMemoryDatabase("API"));
+            var builder = new MySqlConnectionStringBuilder(Configuration.GetConnectionString("DefaultConnection"));
+            builder.Server = Configuration["Server"];
+            builder.Database = Configuration["Database"];
+            builder.UserID = Configuration["Uid"];
+            builder.Password = Configuration["DbPassword"];
```
Comment (wizacass) [inline API/Startup.cs:32]: It is possible to use an initializer like this:
```
var builder = new MySqlConnectionStringBuilder(Configuration.GetConnectionString("DefaultConnection"))
{
    Server = Configuration["Server"],
    Database = Configuration["Database"],
    UserID = Configuration["Uid"],
    Password = Configuration["DbPassword"]
};
```
Maybe ot would be better to change it like this in order to ensure object is created without any errors or missed properties?

File: API/Startup.cs [hunk 5]
```diff
@@ -21,15 +24,27 @@ public void ConfigureServices(IServiceCollection services)
...
 
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env, ApiContext context)
         {
             if (env.IsDevelopment())
             {
                 app.UseDeveloperExceptionPage();
+
+
```
Comment (wizacass) [inline API/Startup.cs:45]: Here shouldn't be 2 lines of spacing,

File: API/Models/SeedData.cs (hunk 3 above)
Comment (lukasrakauskas) [reply API/Models/SeedData.cs:221]: Seeders could be randomized and not hardcoded. As well as have their own class for each table

Comment (Dezash) [reply API/Models/SeedData.cs:221]: As I've mentioned in the description, this is mockup data meant to facilitate testing.

File: API/Models/ApiContext.cs (hunk 1 above)
Comment (Dezash) [reply API/Models/ApiContext.cs]: Fixed.

File: API/Models/Camera.cs (hunk 2 above)
Comment (Dezash) [reply API/Models/Camera.cs]: Fixed.

File: API/Startup.cs (hunk 4 above)
Comment (Dezash) [reply API/Startup.cs]: Yes, thank you for noticing.

File: API/Startup.cs (hunk 5 above)
Comment (Dezash) [reply API/Startup.cs]: Fixed.

File: API/Models/SeedData.cs (hunk 3 above)
Comment (wizacass) [reply API/Models/SeedData.cs:219]: Ok. Wouldn't it be better to create factories for this purpose? Also, this class is wayy too big, consider moving put this code to separate classes as @lukasrakauskas suggested.

Comment (Dezash) [reply API/Models/SeedData.cs]: @wizacass I'd love to have laravel-like model factories here too, but after doing some search, I was not able to find anything similar. If you have any information, do share. For now, I just separated regions into classes for each model.

Comment (wizacass) [reply API/Models/SeedData.cs]: Ok. So for now let's keep it as it is and let's do some research how we can improve on this. Generally, data shouldn't be hardcoded as it is now.

**Pull request merged**